lib/common/imageUtil.py

```diff
@@ -0,0 +1,407 @@
+    return output_sample_fp
+
+
+def convert_to_dct(image_fp, skip_status_bar_fraction=1.0):
+    """
+
+    @param image_fp:
+    @param skip_status_bar_fraction:
+    @return:
+    """
+    dct_obj = None
+    try:
+        img_obj = cv2.imread(image_fp)
+        height, width, channel = img_obj.shape
+        height = int(height * skip_status_bar_fraction) - int(height * skip_status_bar_fraction) % 2
+        img_obj = img_obj[:height][:][:]
+        img_gray = np.zeros((height, width))
+        for channel in range(channel):
+            img_gray += img_obj[:, :, channel]
+        img_gray /= channel
+        img_dct = img_gray / 255.0
+        dct_obj = cv2.dct(img_dct)
+    except Exception as e:
+        logger.error(e)
+    return dct_obj
+
+
+def find_tab_view(file, viewport):
+    """
+
+    @param file:
+    @param viewport:
+    @return:
+    """
+    try:
+        im = Image.open(file)
+        width, height = im.size
+        x = int(math.floor(width / 2))
+        y = int(math.floor(viewport['y'] / 2))
+        pixels = im.load()
+        background = pixels[x, y]
+
+        # Find the top edge
+        x = int(math.floor(width / 2))
+        top = None
+        while top is None and y >= 0:
+            if not colors_are_similar(background, pixels[x, y]):
+                top = y + 1
+            else:
+                y -= 1
+        if top is None:
+            top = 0
+        logger.debug('Browser tab view top edge is {0:d}'.format(top))
+
+        # Find the bottom edge
+        y = int(math.floor(viewport['y'] / 2))
+        bottom = None
+        while bottom is None and y < height:
+            if not colors_are_similar(background, pixels[x, y]):
+                bottom = y - 1
+            else:
+                y += 1
+        if bottom is None:
+            bottom = height
+        logger.debug('Browser tab view bottom edge is {0:d}'.format(bottom))
+
+        tab_view = {'x': viewport['x'], 'y': top, 'width': viewport['width'], 'height': (bottom - top)}
+
+    except Exception as e:
+        tab_view = None
+        logger.error(e)
+
+    return tab_view
+
+
+def find_browser_view(viewport, tab_view):
+    browser_view = {'x': viewport['x'], 'y': tab_view['y'], 'width': viewport['width'],
+                    'height': viewport['y'] + viewport['height'] - tab_view['y']}
+    return browser_view
+
+
+def generate_crop_data(input_target_list, crop_taget_list):
+    """
+    generate crop data for crop images functions
+    @param input_target_list: target obj list, for example {'fp': 'xxxx, 'write_to_file': True}
+    @param crop_taget_list: crop target area list, for example {'viewport': {'x': 2, 'y', ..}}
+    @return: integrate crop data, for example: {'fp_list': [ {'input_fp':xxx, 'output_fp':xxx}], 'crop_area': {'x':2, 'y':3, ..}}
+    """
+    crop_data_dict = {}
+    for input_target in input_target_list:
+        if 'write_to_file' in input_target:
+            if input_target['write_to_file'] is True:
```

Comment:
can't combine these two lines into one line due to attribute could missing in input_target